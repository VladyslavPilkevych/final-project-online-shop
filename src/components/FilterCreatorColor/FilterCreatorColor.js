import React, { memo, useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './FilterCreatorColor.module.scss';
import Button from '../Button/Button';
import {
  addFilterColor,
  removeFilterColor,
} from '../../store/actionCreators/filterAC';

function FilterCreatorColor(props) {
  const { color } = props;
  const dispatch = useDispatch();
  const filterByColor = useSelector((state) => state.filter.filterByColor);
  const [filterColor, setFilterColor] = useState(false);
  useEffect(() => {
    if (filterByColor.length === 0) {
      setFilterColor(false);
    } else if (filterByColor.includes(color)) {
      setFilterColor(true);
    } else {
      setFilterColor(false);
    }
  }, [filterByColor]);
  const toggleColorFn = () => {
    if (!filterColor) {
      dispatch(addFilterColor(color));
    } else {
      dispatch(removeFilterColor(color));
    }
  };
  return (
    <Button
      handleClick={() => { toggleColorFn(); }}
      className={filterColor ? styles.activeFilterColor : styles.colorsItems}
      style={{ backgroundColor: color }}
    />
  );
}

FilterCreatorColor.propTypes = {
  color: PropTypes.string.isRequired,
};

export default memo(FilterCreatorColor);