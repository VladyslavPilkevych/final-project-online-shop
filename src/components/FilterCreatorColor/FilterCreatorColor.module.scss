@import "../../assets/styles/global.module.scss";

@mixin circleColors($border, $width: 30px, $height: 30px, $margin: 3px, $margin-bottom: 7px, $border-radius: 50px, $overflow: hidden, $cursor: pointer) {
    width: $width;
    height: $height;
    margin: $margin;
    margin-bottom: $margin-bottom;
    border-radius: $border-radius;
    overflow: $overflow;
    cursor: $cursor;
    border: $border;
}

.colorsItems {
    // width: 30px;
    // height: 30px;
    // margin: 3px;
    // margin-bottom: 7px;
    // border-radius: 50px;
    // overflow: hidden;
    // cursor: pointer;
    // border: 1px solid $black;
    @include circleColors(1px solid $black);
}
.activeFilterColor {
    @include circleColors(1px solid $black);
    outline-offset: 2px;
    outline: 3px solid $blue;
}