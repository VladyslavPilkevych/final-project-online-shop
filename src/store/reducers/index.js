import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
