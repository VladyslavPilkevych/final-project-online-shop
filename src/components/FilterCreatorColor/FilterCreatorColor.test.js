import { render } from '@testing-library/react';
import FilterCreatorColor from './FilterCreatorColor';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <FilterCreatorColor />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot FilterCreatorColor should render', () => {
  test('Snapshot FilterCreatorColor is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});