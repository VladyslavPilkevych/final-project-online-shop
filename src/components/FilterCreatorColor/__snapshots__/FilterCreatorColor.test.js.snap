// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot FilterCreatorColor should render Snapshot FilterCreatorColor is rendered 1`] = `
<DocumentFragment>
  <button
    class="colorsItems [object Object]"
    type="button"
  />
</DocumentFragment>
`;
